Update extension README and description (#41)
returntocorp/semgrep-vscode: README.md

# Semgrep Visual Studio Code extension

[Semgrep](https://semgrep.dev/) is a fast, static analysis tool for finding bugs, detecting vulnerabilities, and enforcing code standards powered by an open-source engine. Semgrep Visual Studio Code extension scans lines when you change or open files, or scans all files in your workspace.

## Supported languages

See the list of [languages supported by Semgrep OSS Engine](https://semgrep.dev/docs/supported-languages/#semgrep-oss-engine) in Semgrep documentation.

## Prerequisites

> To use Semgrep Visual Studio Code extension, install either pip or Homebrew.

Reviewer: ```suggestion
To use the Semgrep Visual Studio Code extension, install either pip or Homebrew.
```

## Installing Semgrep extension

To successfully install and run the Semgrep Visual Studio Code (VS Code) extension:

1. Install Semgrep locally. See [Installing and running Semgrep locally](https://semgrep.dev/docs/getting-started/#installing-and-running-semgrep-locally) in Semgrep documentation. 
    
    Note: If you install Semgrep through Homebrew, make sure you added Homebrew to your PATH. See [My Mac .apps don’t find Homebrew utilities!](https://docs.brew.sh/FAQ#my-mac-apps-dont-find-homebrew-utilities) in Homebrew documentation.
    
2. Optional: Access the Command Palette by pressing <kbd>Ctrl+⇧Shift+P</kbd> or <kbd>⌘+⇧Shift+P</kbd> (on macOS) and sign in to Semgrep Cloud Platform by running the following command:

    ```
    Semgrep: Sign in
    ```

    After you run the command, click **Open** to open a new tab in your browser, and then click **Activate**.

>     Note: Log in or sign in to improve the performance of Semgrep extension. After log in, you can automatically scan with Semgrep [Pro rules](https://semgrep.dev/docs/semgrep-code/pro-rules/) and add additional rules to the [Rule board](https://semgrep.dev/orgs/-/board) in Semgrep Code.

Reviewer: ```suggestion
    Note: Log in or to improve the performance of Semgrep extension. After logging in, you can automatically scan with Semgrep [Pro rules](https://semgrep.dev/docs/semgrep-code/pro-rules/) and add additional rules from the [Rule board](https://semgrep.dev/orgs/-/board) in Semgrep Code.
```

Suggest to prefer log in because the command is semgrep login.

    
3. Install Semgrep extension in Visual Studio Code. For more information, see [Install an extension](https://code.visualstudio.com/docs/editor/extension-marketplace#_install-an-extension) in Visual Studio Code documentation if necessary.

> 4. In Visual Studio Code, test the extension by hitting <kbd>Ctrl+⇧Shift+P</kbd> or <kbd>⌘+⇧Shift+P</kbd> (macOS) and run the following command:

Reviewer: ```suggestion
4. In Visual Studio Code, test the extension by entering <kbd>Ctrl+⇧Shift+P</kbd> or <kbd>⌘+⇧Shift+P</kbd> (macOS) and run the following command:
```

Author: Oh my mistake, Google Style Guide says that we should use press:
```suggestion
4. In Visual Studio Code, test the extension by pressing <kbd>Ctrl+⇧Shift+P</kbd> or <kbd>⌘+⇧Shift+P</kbd> (macOS) and run the following command:
```

    
    ```
    Semgrep: Scan all files in workspace
    ```

Find results of your scans in Visual Studio Code Problems tab. To display the Problems tab, press <kbd>⇧Shift+Ctrl+M</kbd> or <kbd>⌘+⇧Shift+M</kbd> (macOS).

## Configuring Semgrep extension scans

> Configure Semgrep VS Code extension in extension settings. To access the extension settings:

Reviewer: ```suggestion
Configure Semgrep VS Code extension in the extension settings. To access the extension settings:
```

Author: ```suggestion
Configure Semgrep Visual Studio Code extension in extension settings. To access the extension settings:
```

- **Semgrep › Scan: Configuration**: Specify rules or rulesets you want Semgrep to use to scan your code. Each item can be a YAML configuration file, a URL of a configuration file, or a directory of YAML files. Use "auto" to automatically obtain rules tailored to your project. Your project URL will be used to log in to the Semgrep Registry. See [Running rules](https://semgrep.dev/docs/running-rules/) in Semgrep documentation for more information. When you change the rule configuration, run **Semgrep: Update rules** command in Visual Studio Code Command Palette to update the rules configuration for your next scan.
- **Semgrep › Scan: Include** - List files and directories scanned by Semgrep. This option globally overrides the workspace setting. As a result, Semgrep scans all included paths.
- **Semgrep › Scan: Jobs** - Specify how many parallel jobs of Semgrep extension can run at the same time. The default number of parallel jobs is 1. 

Advanced configuration options: 

- **Semgrep › Scan: Timeout** - After the specified maximum time runs out Semgrep scan times out (stops). Numbers are indicated in seconds. The default value is 30.

> - **Semgre > Trace: Server** - This option is useful for debugging. The messages option displays communication of the Semgrep Visual Studio Code extension with the LSP server. The default option is verbose.

Reviewer: ```suggestion
- **Semgrep > Trace: Server** - This option is useful for debugging. The messages option displays communication of the Semgrep Visual Studio Code extension with the LSP server. The default option is verbose.
```

- `Semgrep: Sign out`: Log out from Semgrep Cloud Platform. Alternatively, you can sign out through your command-line interface by running `semgrep logout`.
- `Semgrep: Scan changed files in a workspace`: Scan files that have been changed since the last commit in your current workspace.
- `Semgrep: Scan all files in a workspace`: Scan all files in the current workspace.
- `Semgrep: Update rules`: For logged-in users. If the rules in the [Rule board](https://semgrep.dev/orgs/-/board) or rules included through **Semgrep › Scan: Configuration** command have been changed, this command loads the new configuration of your rules for your next scan.

Tip: You can also click the Semgrep icon in the Visual Studio Code to quickly access all available commands.

## Features

### Scanning

> Scan your code using Semgrep and get inline results and problem highlighting!

Reviewer: ```suggestion
Scan your code using Semgrep to get inline results and problem highlighting.
```

### Automatic Scanning

> Opened a file? Semgrep scans it right away!

Reviewer: ```suggestion
When you open a file, Semgrep scans it right away.
```

What level of technical writing formality are we applying to this readme? Ignore the suggestion if it's too formal.

Author: @s-santillan I think what you suggest is good. I just felt that somebody who wrote this put a piece of their heart into these words and I should not touch them too much (to avoid causing emotional harm). Let's be more formal so the style is aligned.

### Semgrep App Rules

> Obtain rules configured for your code [Semgrep Code](https://semgrep.dev/products/semgrep-code/)? Login to scan for them!

Reviewer: (See comment above.)

Author: ```suggestion
Sign in to receive better results, new true positive findings, and fewer false positives. When you sign in, you can get all the benefits of [Semgrep Code](https://semgrep.dev/products/semgrep-code/)?
```

### Autofix

> Have an autofix rule? Hit a button and fix it instantly in the Editor.

Author: ```suggestion
With Semgrep autofix rules, you can instantly apply their fix in Visual Studio Code.
```

### Rule Quick Links

Want to go to the definition of a local or Semgrep Code rule? Hover over a match and click the link!

## Support

> Please, if you need our support, join the [Semgrep community Slack workspace](http://go.semgrep.dev/slack) and tell us about any problems you encountered.

Reviewer: Just checking - I thought we don't usually use please? https://developers.google.com/style/tone#politeness-and-use-of-please

Author: Yes, I sort of left it there because I thought that I can sometimes break the rules...

Author: ```suggestion
If you need our support, join the [Semgrep community Slack workspace](http://go.semgrep.dev/slack) and tell us about any problems you encountered.
```

## Supported languages

See the list of [languages supported by Semgrep OSS Engine](https://semgrep.dev/docs/supported-languages/#semgrep-oss-engine) in Semgrep documentation.

## Prerequisites

To use Semgrep Visual Studio Code extension, install either pip or Homebrew.

## Installing Semgrep extension

To successfully install and run the Semgrep Visual Studio Code (VS Code) extension:

1. Install Semgrep locally. See [Installing and running Semgrep locally](https://semgrep.dev/docs/getting-started/#installing-and-running-semgrep-locally) in Semgrep documentation. 
    
    Note: If you install Semgrep through Homebrew, make sure you added Homebrew to your PATH. See [My Mac .apps don’t find Homebrew utilities!](https://docs.brew.sh/FAQ#my-mac-apps-dont-find-homebrew-utilities) in Homebrew documentation.
    
2. Optional: Log in to Semgrep Cloud Platform by running the following command, and then follow the instructions in your command-line interface:

    ```bash
    semgrep login

>     ```

Reviewer 2: We already discussed this on slack, but adding it here again, we should encourage them to use the Semgrep: Login command

Author: Thanks! Implemented in the following commit: https://github.com/returntocorp/semgrep-vscode/pull/41/commits/e081c76ca08ea1c4c777ee864fa3d808ddd9dfc2

    
    Note: Log in or sign in to improve the performance of Semgrep extension. After log in, you can automatically scan with Semgrep [Pro rules](https://semgrep.dev/docs/semgrep-code/pro-rules/) and add additional rules to the [Rule board](https://semgrep.dev/orgs/-/board) in Semgrep Code.
    
3. Install Semgrep extension in Visual Studio Code. For more information, see [Install an extension](https://code.visualstudio.com/docs/editor/extension-marketplace#_install-an-extension) in Visual Studio Code documentation if necessary.
4. In Visual Studio Code, test the extension by hitting <kbd>Ctrl+⇧Shift+P</kbd> or <kbd>⌘+⇧Shift+P</kbd> (MacOS) and run the following command:
    
    ```
    Semgrep: Scan all files in workspace
    ```

> Find results of your scans in Visual Studio Code output. To display the output, press <kbd>⇧Shift+Ctrl+U</kbd> or <kbd>⌘+⇧Shift+U</kbd> (MacOS).

Reviewer 2: The output tab is mostly for debugging, most users probably won't be aware of it

```suggestion
Find results of your scans in Visual Studio Code Problems tab. To display the output, press <kbd>⇧Shift+Ctrl+M</kbd> or <kbd>⌘+⇧Shift+M</kbd> (MacOS).
```

- `Semgrep: Sign in`: Sign in or log in to the Semgrep Cloud Platform (this command opens a new window in your browser). When you sign in, you can automatically scan with Semgrep [Pro rules](https://semgrep.dev/docs/semgrep-code/pro-rules/) and add additional rules to the [Rule board](https://semgrep.dev/orgs/-/board) in Semgrep Code. If you are logged in with the command-line interface using `semgrep login` you are already signed in with the Visual Studio Code Semgrep extension also.
- `Semgrep: Sign out`: Log out from Semgrep Cloud Platform.
- `Semgrep: Scan changed files in a workspace`: Scan files that have been changed since the last commit in your current workspace.
- `Semgrep: Scan all files in a workspace`: Scan all files in the current workspace.
- `Semgrep: Update rules`: For logged-in users. If the rules in the [Rule board](https://semgrep.dev/orgs/-/board) have been changed, this command loads the new configuration of your rules for your next scan.

Tip: You can also click the Semgrep icon in the Visual Studio Code to quickly access all available commands.

> ## Features

Reviewer 2: I don't have access to your fork to change this, so can you remove these sections:

- Automatic Config Detection
- Hot Reloading
- Metavariable Labelling

Author: Thanks! Implemented in: https://github.com/returntocorp/semgrep-vscode/pull/41/commits/c1731bb99c5ba534a811de9f8d800c1d5f09ea6a

Want to go to the definition of a local or app rule? Hover over a match and click the link!

## Configuration

> You can set the following options by going to Preferences > Settings:

Reviewer 2: Let's just delete this section, since we already go over the important ones of these above

Author: Implemented in: https://github.com/returntocorp/semgrep-vscode/pull/41/commits/c1731bb99c5ba534a811de9f8d800c1d5f09ea6a
